### PR TITLE
qemu/tests: Add new case without_kvmclock for multi host migration

### PR DESCRIPTION
**Pull request by spcui**
1. Add step for checking clocksource in guest.
2. Add new case without_kvmclock.

Signed-off-by: Shuping Cui <scui@redhat.com>

ID: 1298771

### CODE REVIEW

File: qemu/tests/cfg/multi_host.cfg [hunk 1]
```diff
@@ -87,6 +87,7 @@
                     reboot_method = system_reset
                     type = migration_multi_host_with_reboot
                 - timedrift:
+                    only Linux
```
Comment (CongSmile) [inline qemu/tests/cfg/multi_host.cfg]: @spcui 

Could you help explain why Linux only?

File: qemu/tests/migration_multi_host_timedrift.py [hunk 2]
```diff
@@ -55,12 +56,32 @@ def __init__(self, test, params, env):
...
+                    current_clocksource = session.cmd(check_clocksource_cmd)
+                    logging.debug("current_clocksource in guest is: '%s'" %
+                                  current_clocksource)
+                    if clocksource == "kvm-clock":
+                        s = "kvm-clock" in current_clocksource
+                    else:
+                        s = "kvm-clock" not in current_clocksource
+                    if not s:
+                        raise error.TestFail("Guest didn't use '%s' "
+                                             "clocksource" % clocksource)
```
Comment (CongSmile) [inline qemu/tests/migration_multi_host_timedrift.py]: Why failed the case if no kvmclock? I think a warning is fine.

File: qemu/tests/cfg/multi_host.cfg (hunk 1 above)
Comment (spcui) [inline qemu/tests/cfg/multi_host.cfg]: kvm-clock only for linux guest.

File: qemu/tests/migration_multi_host_timedrift.py (hunk 2 above)
Comment (spcui) [inline qemu/tests/migration_multi_host_timedrift.py]: if boot with -cpu ,-kvmclock, the clocksource in guest should be not kvmclock, if not, maybe it's qemu bug.

File: qemu/tests/migration_multi_host_timedrift.py [hunk 3]
```diff
@@ -55,12 +56,32 @@ def __init__(self, test, params, env):
...
+                                  logging.info)
+                    check_clocksource_cmd = params.get("check_clocksource_cmd")
+                    clocksource = params.get("clocksource", "kvm-clock")
+                    current_clocksource = session.cmd(check_clocksource_cmd)
+                    logging.debug("current_clocksource in guest is: '%s'" %
+                                  current_clocksource)
+                    if clocksource == "kvm-clock":
+                        s = "kvm-clock" in current_clocksource
+                    else:
+                        s = "kvm-clock" not in current_clocksource
```
Comment (suqinhuang) [inline qemu/tests/migration_multi_host_timedrift.py]: The logic is strange,
why not compare clocksource with current_clocksource directly.

File: qemu/tests/migration_multi_host_timedrift.py [hunk 4]
```diff
@@ -55,12 +56,32 @@ def __init__(self, test, params, env):
...
             for vm in mig_data.vms:
                 session = vm.wait_for_login()
+
+                if self.is_src and params.get("os_type") == "linux":
+                    error.context("Check the clocksource in guest.",
+                                  logging.info)
+                    check_clocksource_cmd = params.get("check_clocksource_cmd")
+                    clocksource = params.get("clocksource", "kvm-clock")
+                    current_clocksource = session.cmd(check_clocksource_cmd)
+                    logging.debug("current_clocksource in guest is: '%s'" %
```
Comment (suqinhuang) [inline qemu/tests/migration_multi_host_timedrift.py]: need to use logging.info

File: qemu/tests/migration_multi_host_timedrift.py (hunk 3 above)
Comment (spcui) [inline qemu/tests/migration_multi_host_timedrift.py]: for different guest, current_clocksource can return the different value without kvm-clock

**Comment by suqinhuang** [comment]
Acked-by: Suqin Huang

**Comment by CongSmile** [comment]
Acked-by: Cong Li <coli@redhat.com>

**Pull request merged**